### PR TITLE
S7 Filtering

### PR DESCRIPTION
**Pull request by kuzivka**

### CODE REVIEW

File: src/components/common/StarRating.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+`;
+function Rating({
+  value,
+  text,
+  amount = 5,
+  color = '#ffc600',
+}: IRating): JSX.Element {
+  return (
+    <Box display="flex" alignItems="center">
+      <ul style={{ display: 'flex', flexDirection: 'row', gap: '2px' }}>
```
Comment (Yuliia143) [inline src/components/common/StarRating.tsx]: inline styles

File: src/components/common/StarRating.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        {Array.from({ length: amount }, (_, i) => i + 1).map((number) => (
+          <li key={number} style={{ display: 'inline' }}>
+            <StyledSpan color={color}>
+              {Math.ceil(value) >= number ? <Star /> : <StarOutline />}
```
Comment (Yuliia143) [inline src/components/common/StarRating.tsx]: logic in JSX

File: src/components/filters/RateFilter.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    dispatch(setRateFilter(checkedRate));
+  }, [checkedRate, dispatch]);
+
+  return (
+    <Box sx={filterContainer}>
+      <Typography variant="h5" sx={filterLabel}>
+        Ratings
+      </Typography>
+      <List sx={filterList}>
+        {RATES.sort((a, b) => b - a).map((rate) => (
```
Comment (Yuliia143) [inline src/components/filters/RateFilter.tsx]: logic in JSX

File: src/components/common/StarRating.tsx (hunk 1 above)
Comment (Kost927) [inline src/components/common/StarRating.tsx]: .

File: src/styles/theme.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { createTheme } from '@mui/material/styles';
+
+export const theme = createTheme({
+  palette: {
+    primary: {
+      main: '#6A983C',
```
Comment (Yuliia143) [inline src/styles/theme.ts]: where should colors be?

File: src/components/filters/BrandFilter.tsx [hunk 5]
```diff
@@ -1,3 +1,93 @@
...
+    (state: RootState) => state.productList.categoryFilter
+  );
+  const [checkedFarms, setCheckedFarms] = useState(farmsToFilter);
+
+  const checkboxChangeHandler = useCallback(
+    (farmId: number) => (event: ChangeEvent<HTMLInputElement>) => {
+      const farmsSet = new Set(checkedFarms);
+      if (event.target.checked) {
+        setCheckedFarms(Array.from(farmsSet.add(farmId)));
+      } else if (!event.target.checked) {
```
Comment (Kost927) [inline src/components/filters/BrandFilter.tsx]: use if (checked) {...} else {...}.

File: src/components/filters/RateFilter.tsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+  );
+
+  const [checkedRate, setCheckedRate] = useState<number[]>(ratesToFilter);
+
+  const checkboxChangeHandler = useCallback(
+    (farmId: number) => (event: ChangeEvent<HTMLInputElement>) => {
+      const rateSet = new Set(checkedRate);
+      if (event.target.checked) {
+        setCheckedRate(Array.from(rateSet.add(farmId)));
+      } else if (!event.target.checked) {
```
Comment (Kost927) [inline src/components/filters/RateFilter.tsx]: same

**Review by Yuliia143 — APPROVED** [review]

**Pull request merged**